src/in-depth/signals-channels.rs

```diff
@@ -0,0 +1,31 @@
+use std::time::Duration;
+use crossbeam_channel::{bounded, tick, Receiver, select};
+
+fn ctrl_channel() -> Result<Receiver<()>, ctrlc::Error> {
+    let (sender, receiver) = bounded(100);
+    ctrlc::set_handler(move || {
+        let _ = sender.send(());
+    })?;
+
+    Ok(receiver)
+}
+
+fn main() -> Result<(), exitfailure::ExitFailure> {
+    let ctrl_c_events = ctrl_channel()?;
+    let ticks = tick(Duration::from_secs(1));
+
+    loop {
+        select! {
```

Comment:
`select!` is deprecated, isn't it?

Reply:
This is the select macro from crossbeam-channels, which I think is still going strong

Reply:
This is `crossbeam_channel::select!`, not `std::select!` (which *is* deprecated).

Reply:
ah, pardon me. :)